### ISSUE TITLE
Upgrade Typescript to v4

### ISSUE THREAD

**Issue by lmm-git** [quasarframework/quasar#8544]
**Is your feature request related to a problem? Please describe.**
Typescript Version 3 is used with Quasar

**Describe the solution you'd like**
Upgrade to Version 4

**Describe alternatives you've considered**
-

**Additional context**
Version 4 was released September of last year. I actually don't know exactly whether this could break anything, but I would really appreciate the availability of newer TypeScript features in Quasar.

**Comment by rstoenescu** [comment]
Hi, yes we are planning it for Quasar v2 already.

**Comment by IlCallo** [comment]
Quasar v2 already support TS4.1, upgrading to TS4.2 right now.

Opening a PR for TS4.2 support on v1 too

**Comment by rstoenescu** [comment]
Will be available in "@quasar/app" v2.2.0.

**Comment by IlCallo** [comment]
Leaving it here for who's upgrading from an existing Quasar codebase.
You shall upgrade you local project deps accordingly to use TS4 at its fullest:
```
"@vue/composition-api": "^1.0.0-rc.3",
"eslint": "^7.21.0",
"eslint-plugin-vue": "^7.7.0",
"eslint-config-standard": "^16.0.2",
"eslint-config-prettier": "^8.1.0",
"@typescript-eslint/eslint-plugin": "^4.16.1",
"@typescript-eslint/parser": "^4.16.1",
```

If you're using Prettier, you can remove `'prettier/@typescript-eslint', 'prettier/vue'` from your ESLint rules
